### PR TITLE
introduce MAX_MAPLIST_COLS

### PR DESCRIPTION
**Pull request by illwieckz**
After #964, I thought it would be useful to also set a `MAX_MAPLIST_COLS` constant to make code easier to read and modify. Also, one day we can imagine to substitute this constant and the ROWS one by variables relative to the screen size or things like that.

So currently it's just another constant in code so we know what is what. By the way, I have a question: what's the purpose of the other hardcoded value set to `2`? Is it `MAX_MAPLIST_COLS - 1`? Is it always `2`? And Why?

Of course if that `2` is relative to `MAX_MAPLIST_COLS` please not merge this change before engraving this relation in code!

_Edit: it is! It's effectively `MAX_MAPLIST_COLS - 1`, updated the commit accordingly._

### CODE REVIEW

**Comment by DolceTriade** [comment]
LGTM. It's not clear what the extra +2 is for. If I had to guess, it seems it's for formatting purposes, as if they were printing two additional maps, but they aren't printing anything else. Does deleting it make stuff look funky?

**Comment by illwieckz** [comment]
Yeah, it's `MAX_MAPLIST_COLS - 1`. It's to be sure to fill an extra row when the number of map is not a multiple of `MAX_MAPLIST_COLS` and `MAX_MAPLIST_ROWS`, and `1` being the minimal number of map to trigger the filling (because `1 extra map + MAX_MAPLIST_COLS - 1 = MAX_MAPLIST_COLS` so `N extra map + MAX_MAPLIST_COLS - 1 > MAX_MAPLIST_COLS`, the `max( 1, ( rows + MAX_MAPLIST_ROWS - 1 ) / MAX_MAPLIST_ROWS )` code is doing that `>` test implicitely.

Without that, any line with less columns than `MAX_MAPLIST_COLS` will no be counted and displayed.

With 33 maps with 2 rows per page and 4 columns per row, removing that "extra +2" shows this:

```
]/listmaps 4
 test                 transit              uncreation           vega                
 thunder              tremor               usstremor            western             
 
listmaps: listing 8 of 33 maps 
listmaps: page 4 of 4 
```

4 page means 32 maps, so the 5th page for the remaining map (here, yocto) is missing and can't be asked for display.

With 33 maps with 3 rows per page and 4 columns per row, using `MAX_MAPLIST_COLS - 1` instead of `2` shows this:

```
]/listmaps 4
 test                 transit              uncreation           vega                
 thunder              tremor               usstremor            western             
 
listmaps: listing 8 of 33 maps 
listmaps: page 4 of 5; use 'listmaps 5' to see more 
]/listmaps 5
 yocto               
 
listmaps: listing 1 of 33 maps 
listmaps: page 5 of 5 
```

**Comment by illwieckz** [comment]
Updated PR.

**Comment by DolceTriade** [comment]
Cool, nice explanation.

File: src/sgame/sg_cmds.cpp [hunk 1]
```diff
@@ -3853,19 +3854,19 @@ void Cmd_ListMaps_f( gentity_t *ent )
 			   }
 	);
 
-	int rows = ( mapNamesCount + 2 ) / 3;
+	int rows = ( mapNamesCount + MAX_MAPLIST_COLS - 1 ) / MAX_MAPLIST_COLS;
```
Comment (DolceTriade) [inline src/sgame/sg_cmds.cpp]: Please add a comment here about why the MAX_MAPLIST_COLS - 1 is necessary.

Comment (illwieckz) [reply src/sgame/sg_cmds.cpp]: done

**Pull request merged**